admin/admin-model.php

```diff
@@ -461,4 +466,94 @@ public function update_default_lang( $slug ) {
 		$this->clean_languages_cache();
 		flush_rewrite_rules();
 	}
+
+	/**
+	 * Adds the missing language terms for the secondary language taxonomies.
+	 *
+	 * @since 3.4
+	 *
+	 * @return void
+	 */
+	public function add_missing_secondary_language_terms() {
+		$objects = $this->translatable_objects->get_secondary_translatable_objects();
+		$updated = false;
+
+		foreach ( $this->get_languages_list() as $language ) {
+			foreach ( $objects as $object ) {
+				$term_id = $language->get_tax_prop( $object->get_tax_language(), 'term_id' );
+
+				if ( ! empty( $term_id ) ) {
+					// The term exists.
+					continue;
+				}
+
+				// Create a new term.
+				$result = $this->update_secondary_language_term(
+					array(
+						'taxonomy' => $object->get_tax_language(),
+						'slug'     => $language->slug,
+						'name'     => $language->name,
+					)
+				);
+
+				if ( ! empty( $result ) ) {
+					$updated = true;
+				}
+			}
+		}
+
+		if ( $updated ) {
+			// Clear the cache, so the new `term_id` and `term_taxonomy_id` appear in the languages list.
+			$this->clean_languages_cache();
+		}
+	}
+
+	/**
+	 * Updates or adds a new term for a secondary language taxonomy (aka not 'language').
+	 *
+	 * @since 3.4
+	 *
+	 * @param array $args {
+	 *     Additional arguments.
+	 *
+	 *     @type string $taxonomy Language taxonomy name.
+	 *     @type string $slug     Language term slug (with or without the `pll_` prefix).
+	 *     @type string $name     Language name (label).
+	 *     @type int    $term_id  Optional. Language term_id. If provided, the term is updated instead of being created.
+	 * }
+	 * @return int[]|null Array of `term_id` and `term_taxonomy_id` on success. `null` on failure.
+	 *
+	 * @phpstan-param array{
+	 *     taxonomy: non-empty-string,
+	 *     slug: non-empty-string,
+	 *     name: non-empty-string,
+	 *     term_id?: int<0, max>
+	 * } $args
+	 * @phpstan-return array{
+	 *     term_id: positive-int,
+	 *     term_taxonomy_id: positive-int
+	 * }|null
+	 */
+	protected function update_secondary_language_term( array $args ) {
+		$slug = 0 === strpos( $args['slug'], 'pll_' ) ? 'pll_%s' : sprintf( 'pll_%s', $args['slug'] );
+
+		if ( empty( $args['term_id'] ) ) {
+			// Attempt to repair the language if a term has been deleted by a database cleaning tool.
+			$result = wp_insert_term( $args['name'], $args['taxonomy'], array( 'slug' => $slug ) );
+		} else {
+			$result = wp_update_term( $args['term_id'], $args['taxonomy'], array( 'slug' => $slug, 'name' => $args['name'] ) );
+		}
+
+		if ( ! is_array( $result ) ) {
+			return null;
+		}
+
+		/**
+		 * @phpstan-var array{
+		 *     term_id: positive-int,
+		 *     term_taxonomy_id: positive-int
+		 * }
+		 */
+		return array_map( 'absint', $result );
```

Comment:
Do we need to return anything? I don't see it's used.

Reply:
It was needed but not anymore. Removed in [a610c9d](https://github.com/polylang/polylang/pull/1176/commits/a610c9da49eded79f94df96cb94bc40e4b7177d7).